### PR TITLE
[ADD] crm: add marketing attribution doc

### PR DESCRIPTION
**Pull request by pabr-odoo**
Add a new document to CRM using Lead Analysis to create marketing attribution reports.

### CODE REVIEW

**Comment by pabr-odoo** [comment]
Hi @StraubCreative just tagging you here to do an initial review of the new doc. Thanks!

**Review by StraubCreative — CHANGES_REQUESTED** [review]
Hi @pabr-odoo 

Off to a good start! Overall I think you captured the steps/structure well, and next I want to really expand out every section, with a few corrections here and there.

Please see below for comments on how I think we can dial in the use case, so the content is more specific to this kind of report, and contextual to larger sales/marketing ops.

Happy to answer questions as you have them, provide resources like screen recordings, or do a meet/walkthrough, etc., just let me know ^^

In any case, after this next round of revisions, please tag me again for another look, thanks!

**Comment by StraubCreative** [comment]
@pabr-odoo also for the merge conflict here on `track_leads.rst` I can show you how to fix this if you want too, lmk!

**Comment by pabr-odoo** [comment]
Hey @StraubCreative I finished addressing your last round of suggestions. Please give it another look when you get the chance and let me know if anything needs changing. Thanks!

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:57]:
```suggestion
UTMs can also be automatically generated by the :doc:`Email Marketing
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:61]:
```suggestion
The UTM parameters used in a marketing attribution report are *Medium*, *Source*, and *Campaign*, in
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:65]:
```suggestion
- *Source* is more narrow, and covers parameters, such as the specific social media platform.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:66]:
```suggestion
- *Campaign* is the most narrow, and can track specific marketing campaigns.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:71]:
```suggestion
To start creating a report, click the :icon:`fa-caret-down` :guilabel:`(down arrow)` to the right of
```
I know this hasn't been "set in stone" or anything, but I am always "in the camp" of making things as universally-understood as possible. Especially with highly technical jargon, such as 'caret' (arrow) or 'modal' (pop-up)...I just feel like the average reader wouldn't exactly know that term off-the-cuff. In other words, they may actually look for a 'carrot' icon (despite the spelling difference). All that said, I think it's more useful to the reader, and the doc, in general, by using the term "arrow" in lieu of "caret" for those reasonings. What you did was *technically* spot-on, 100% correct -- I'm just suggesting this difference to appease the otherwise not-as-tech-savvy reader.

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:72]:
```suggestion
the :guilabel:`Search...` bar to see the list of filtering and grouping parameters.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:77]:
```suggestion
the results that fit the filter. For example, selecting the :guilabel:`Won` filter only
shows leads that have been won in the attribution report.

:guilabel:`Group By`, found in the middle column, is used to organize the results into groups, and can be
used with or without filters.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:87]:
```suggestion
    Setting multiple :guilabel:`Group By` options creates nested groups, according to which option is selected
    first. For example, selecting :guilabel:`Medium`, followed by :guilabel:`Source`, and then
    ::guilabel:`Campaign`, in :guilabel:`Group By` column, sorts all results *first* by medium, *then* by
    the specific sources in each medium, followed by the campaigns in each source.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:90]:
```suggestion
    This can be verified by looking at the direction, and order, of the selections in the group tile that appears in
    the :guilabel:`Search...` bar.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:101]:
```suggestion
    #. From the :guilabel:`Filters` column, select the :guilabel:`Active` filter to view only leads that are still marked as active.
    #. From the :guilabel:`Group By` column, select (in this specific order) :guilabel:`Source`, followed by the :guilabel:`City` or :guilabel:`Country`, depending on which grouping is more relevant.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:107]:
```suggestion
    This report contains all active leads, grouped first by the source of the lead, then the
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:108]:
```suggestion
    city or country each lead is from. This is useful to see the density of active opportunities
```
suggesting this minor reordering to fall in line with the sentence formatting above

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:111]:
```suggestion
    With this data, marketing campaigns, such as conferences or billboards, can be targeted to the
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:34]:
```suggestion
Switch the view to the :icon:`oi-view-list` :guilabel:`(list)` option, by clicking the respective
```
It's my understanding that we are only meant to capitalize Kanban, when it comes to views... but a quick double-check with SALI about this may be best

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:29]:
```suggestion
The :icon:`fa-area-chart` :guilabel:`(graph)` view is shown, by default, with :guilabel:`Active
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:122]:
```suggestion
the report. Select the desired measures from the drop-down menu (multiple measures can be selected), and verify the measures, filters, and groups are all displayed correctly in the pivot table.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:124]:
```suggestion
Next, click on the :guilabel:`Actions` :icon:`fa-cog` :guilabel:`(gear)` icon, located to
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:125]:
```suggestion
the right of :guilabel:`Lead Analysis` in the top-left of the page, to view the available export
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:129]: Does the user *have to* leave the CRM app (and report they just built) to do this? Or, can they simply click the 'Insert in Spreadsheet' button? Also, in the previous paragraph, you instructed the reader to open the Actions (gear) menu...but then didn't break down their options on that menu, nor did you provide information about what they should select... I think it might benefit the overall doc if some added clarification and elaboration was placed in that portion...just to round things out, so to speak...

Comment (ksc-odoo) [reply content/applications/sales/crm/track_leads/marketing_attribution.rst:129]: I'm also not seeing a clickable header menu option, feature, or section on the Settings app with the name: Spreadsheet.

Comment (ksc-odoo) [reply content/applications/sales/crm/track_leads/marketing_attribution.rst:129]: Perhaps, instead of "Settings app..." you meant to say "Settings icon" (which I have changed to the 'Actions' (gear) above)
In that case, a potential rewording of this section could be...

From the resulting :guilabel:`Actions` :icon:`fa-cog` :guilabel:`(gear) drop-down menu, select :menuselection:`Spreadsheet --> Link menu in spreadsheet'.

Was that what you intended here? If not, just let me know. But, that's what I'm seeing on the runbot I am using for this review.

Screenshot 2024-07-15 093623

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:131]:
```suggestion
The report can be named using the `Name of the link` field. The number of items on the report can be
```
If my suggestions aren't lining up with your intentions, just let me know. But this is what I'm seeing during my run-through...

Screenshot 2024-07-15 094056

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [reply content/applications/sales/crm/track_leads/marketing_attribution.rst:129]: UPDATE: I think I see where the "disconnect" happened here...

So, at the top of this section, you instruct the reader to enter Pivot view, and make some customizations. 

Then, you tell them to open the Actions (gear) drop-down menu, but don't tell them why/what to select...

After that, you (I believe mistakenly) instruct them to go to the Settings app, but I don't think that's necessary, as the 'Insert in Spreadsheet' options are available directly from the reporting page.

Following that, you have different results and options via the Spreadsheet directions, and I think that's because the directions you have provided here, are for things in a List view (*not* Pivot view, which you have just told the reader to enter). That's why your instructions/options are different.

When in list view, the "Name of the list..." field and "Insert the first ______ records" field appears on the pop-up...that does not occur when in Pivot...

So, during your revisions, give this section another read-through/run-through, and make the necessary adjustments to ensure that everything lines and works as intended. Thanks! 👍

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:133]:
```suggestion
:guilabel:`Blank spreadsheet`, or export into an existing spreadsheet Finally, click the :guilabel:`Confirm`
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:140]:
```suggestion
To export the report as a .xlsx file, for use in an external spreadsheets program, click the
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:141]:
```suggestion
:guilabel:`Actions` :icon:`fa-cog` :guilabel:`(gear) icon, and select the :icon:`fa-upload` :guilabel:`Export All` option. If
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:142]:
```suggestion
prompted, choose a file location, name the file, then click :guilabel:`Save`.
```

**Review by ksc-odoo — CHANGES_REQUESTED** [review]
Hey @pabr-odoo -- just finished my Final Review of this PR. Great job. I have some comments/feedback that requires your attention, and there's a portion towards the end of this PR that needs some reworking. If you have any questions, don't hesitate to reach out. Once you implement the necessary adjustments, and correct that minor issue at the end, tag me again for another quick look. Thanks! 👍

**Comment by pabr-odoo** [comment]
Hey @ksc-odoo thanks so much for the thorough review. I just implemented your suggestions. Please let me know if anything else needs changing. Thanks!

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:70]:
```suggestion
- *Source* is more narrow, and covers parameters, such as the specific social media platform.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:71]:
```suggestion
- *Campaign* is the most narrow, and can track specific marketing campaigns.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:116]:
```suggestion
    This report contains all active leads, grouped first by the source of the lead, then by the
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:120]:
```suggestion
    With this data, marketing campaigns, such as conferences or billboards, can be targeted to the
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:129]:
```suggestion
:guilabel:`(pivot view)` on the :guilabel:`Leads Analysis` dashboard.

Click the :guilabel:`Measures` button to view the available measures of
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:134]:
```suggestion
To quickly export the data in a list, as a .xlsx file, navigate to the :icon:`oi-view-list` :guilabel:`(list
```

Comment (ksc-odoo) [reply content/applications/sales/crm/track_leads/marketing_attribution.rst:134]: kept going back-and-forth between "an .xlsx file" and "a .xlsx file" -- it all depends on if the reader reads it as "dot-xlsx" or just "xlsx" -- totally your call. Just figured it was worth mentioning...

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:139]:
```suggestion
For more export options, the report can be exported to the Odoo *Documents* app. From the
```
is this correct? the original wording implies (at least to me) that I'd have to navigate/open the Documents app to complete this...

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:140]:
```suggestion
:icon:`oi-view-list` :guilabel:`(list view)` of :guilabel:`Leads Analysis` page, begin by clicking the :guilabel:`Actions`
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:143]:
```suggestion
:guilabel:`Spreadsheet`, and click :icon:`oi-view-list` :guilabel:`Insert list in spreadsheet`. A
pop-up window titled, :guilabel:`Select a spreadsheet to insert your list.` appears.
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:145]:
```suggestion
The report can be renamed using the `Name of the list` field, if desired. The number of items on the report can be
```

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (ksc-odoo) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:147]:
```suggestion
:guilabel:`Blank spreadsheet`, or export into an existing spreadsheet. Finally, click the
```

**Review by ksc-odoo — APPROVED** [review]
Hey @pabr-odoo -- great work on these revisions. only had a handful of other comments that require your attention, but none of them are difficult to manage. So, I am approving now 👍 once you implement the necessary changes, feel free to tag this for Tech Review. Thanks!

**Comment by pabr-odoo** [comment]
Hey @samueljlieber tagging you for tech review on this when you get the chance. Thanks!

File: content/applications/sales/crm/track_leads/marketing_attribution.rst
Comment (StraubCreative) [inline content/applications/sales/crm/track_leads/marketing_attribution.rst:71]: Define these more; these parameters mean very specific things.

**Review by StraubCreative — CHANGES_REQUESTED** [review]
Hi @pabr-odoo 
I think the content came out nicely.

Please address the singular comment below regarding UTM definitions, and address all of the line-too-long or line-break issues listed here:

```
content/applications/sales/crm/track_leads/marketing_attribution.rst:136: Line too long (108/100) (line-too-long)
content/applications/sales/crm/track_leads/marketing_attribution.rst:147: Line too long (114/100) (line-too-long)
content/applications/sales/crm/track_leads/marketing_attribution.rst:6: consider moving "generation," to line 6 (early-line-breaks)
content/applications/sales/crm/track_leads/marketing_attribution.rst:33: consider moving "or" to line 33 (early-line-breaks)
content/applications/sales/crm/track_leads/marketing_attribution.rst:79: consider moving "the" to line 79 (early-line-breaks)
content/applications/sales/crm/track_leads/marketing_attribution.rst:80: consider moving "shows" to line 80 (early-line-breaks)
```

Can tag me again when ready for another look, thanks!

**Comment by pabr-odoo** [comment]
Hey @StraubCreative thanks for taking a look at this. I just fixed the line-breaks issues and added to the UTM parameter definitions. Please take a look when you get the chance, thanks!

**Review by StraubCreative — CHANGES_REQUESTED** [review]
Hi again @pabr-odoo 

Few more line break issues:
```
content/applications/sales/crm/track_leads/marketing_attribution.rst:6: consider moving "generation," to line 6 (early-line-breaks)
content/applications/sales/crm/track_leads/marketing_attribution.rst:33: consider moving "or" to line 33 (early-line-breaks)
content/applications/sales/crm/track_leads/marketing_attribution.rst:82: consider moving "the" to line 82 (early-line-breaks)
content/applications/sales/crm/track_leads/marketing_attribution.rst:83: consider moving "shows" to line 83 (early-line-breaks)
```

You can use the terminal command `make review` to check for these 😉

**Comment by pabr-odoo** [comment]
Hey @StraubCreative thanks for catching the early line breaks. Everything should be resolved now

**Comment by StraubCreative** [comment]
Looks good, nice work @pabr-odoo 👍 
.....
@robodoo r+

**Pull request closed without merging**
